test/built-ins/Array/from/source-array-boundary.js

```diff
@@ -0,0 +1,32 @@
+// Copyright 2015 Microsoft Corporation. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+description: Source array with boundary values
+es6id: 22.1.2.1
+---*/
+
+var array = [ Number.MAX_VALUE, Number.MIN_VALUE, Number.NaN, Number.NEGATIVE_INFINITY, Number.POSITIVE_INFINITY ];
+var mapObj = {
+    arrayIndex : -1,
+    mapFn(value, index) {
+        this.arrayIndex++;
+        if (index !== 2) {
+            assert.sameValue(value, array[this.arrayIndex], "Value mismatch in mapFn at index " + index + ".");
+        } else {

```

Comment:
This branch is unnecessary. The assertion in the `if` block can be made for every iteration, so the `if..else` statement can be replaced with the body of the `if` block.